src/components/MultiStepIndicator/components/StepIndicator/StepIndicator.jsx

```diff
@@ -90,11 +105,13 @@ const StepIndicator = ({
   return (
     <li
       className={cx(...getClassNamesWithSuffix(""), stepComponentClassName, {
-        [baseClassNameWithAnimation]: statusChangeAnimationState
+        [baseClassNameWithAnimation]: statusChangeAnimationState,
+        clickable: onClick
       })}
       aria-label={ariaLabel}
+      onClick={handleClick}
     >
-      <div className={cx(...getClassNamesWithSuffix("__number-container"))} ref={componentRef}>
+      <div className={cx(...getClassNamesWithSuffix("__number-container"))} ref={componentRef} tabIndex="0">
```

Comment:
please add role="button" or use useButton from react aria - it should also take care the useKeyEvents